R/lint.R

```diff
@@ -335,10 +340,10 @@ validate_linter_object <- function(linter, name) {
     return(linter)
   }
   if (!is.function(linter)) {
-    stop(gettextf(
-      "Expected '%s' to be a function of class 'linter', not a %s of class '%s'",
-      name, typeof(linter), class(linter)[[1L]]
-    ), call. = FALSE)
+    cli_abort(c(
+      i = "Expected {.fn {name}} to be a function of class {.cls linter}.",
+      x = "Instead, it is {.obj_type_friendly {linter}}."
```

Comment:
This gives us a {cli} version dependency right?

https://github.com/r-lib/cli/blob/main/NEWS.md#cli-340

Reply:
Good point! I have updated DESCRIPTION to reflect this.